Framework/Core/src/DataRelayer.cxx

```diff
@@ -196,6 +196,14 @@ void sendVariableContextMetrics(VariableContext& context, TimesliceSlot slot,
 DataRelayer::RelayChoice
   DataRelayer::relay(std::unique_ptr<FairMQMessage>&& header,
                      std::unique_ptr<FairMQMessage>&& payload)
+{
+  return relay(std::move(header), &payload, 1);
```

Comment:
With the logic implemented below this should be zero as it describes the number of additional payloads. It does not harm since there is a division by 2, which make it to zero then. I would leave it like this and change the implementation below so that numPayloads denotes all payloads.

Reply:
I fixed the logic to have `(restOfPartsSize - 1) / 2`. Also `numPayloads` was renamed to `restOfPartsSize` which is the total size of all the messages part of the multipart, minus the first header which is passed separately.

Reply:
perfect